docs/upgrading.md

```diff
@@ -4,6 +4,307 @@ description: Next.js のアップグレード方法を学びます。
 
 # アップグレードガイド
 
+## 11から12へのアップグレード
+
+### Node.jsの最小バージョン
+
+Node.js の最小バージョンは 12.0.0 から 12.22.0 に変更されました。
+
+### Reactの最新版へのアップグレード
+
+アップグレードするには、以下のコマンドを実行します。
+
+```
+npm install react@latest react-dom@latest
+```
+
+yarn を使用する場合:
+
+```
+yarn add react@latest react-dom@latest
+```
+
+### Next.jsのバージョンを12にアップグレードする
+
+バージョンアップするには、ターミナルで以下のコマンドを実行します:
+
+```
+npm install next@12
+```
+
+または:
+
+```
+yarn add next@12
+```
+
+### Babelに代わるSWC
+
+Next.js は、JavaScript/TypeScript のコンパイルに Rust ベースのコンパイラ[SWC](https://swc.rs/)を使用するようになりました。
+この新しいコンパイラは、個別のファイルをコンパイルする際に Babel より最大 17 倍、Fast Refresh では最大 5 倍高速になります。
+
+Next.js は、[カスタムBabel設定](https://nextjs.org/docs/advanced-features/customizing-babel-config)を持つアプリケーションとの完全な後方互換性を提供します。
+Next.js がデフォルトで処理する styled-jsx や、`getStaticProps` / `getStaticPaths` / `getServerSideProps`のツリーシェイクなどの変換は、すべて Rust に移植されました。
+
+アプリケーションが Babel の設定をカスタマイズしている場合、Next.js は自動的に JavaScript/Typescript のコンパイルに SWC を使わず、Next.js 11 と同じように Babel を使うようにフォールバックされます。
+
+現在、カスタム Babel 変換を必要とする外部ライブラリとの統合の多くは、近い将来で Rust ベースの SWC 変換に移植される予定です。これには以下が含まれますが、これらに限定されるものではありません:
+
+- Styled Components
+- Emotion
+- Relay
+
+SWC を採用するのに役立つ変換を優先するため、[フィードバックスレッド](https://github.com/vercel/next.js/discussions/30174)で `.babelrc` を提供してください。
+
+### SWCによるTerserの最小化の置き換え
+
+`next.config.js`に以下のフラグを追加することで Terser を SWC に置き換えて、JavaScript の minify を最大 7 倍高速化出来ます。:
+
+```js
+module.exports = {
+  swcMinify: true,
+}
+```
+
+SWC による最小化は、Next.js 12.1 でデフォルトになる前により多くの実際の Next.js アプリケーションでテストできるようオプトインのフラグになっています。
+最小化についてのフィードバックがあれば、[フィードバックスレッド](https://github.com/vercel/next.js/discussions/30237)に残してください。
+
+### styled-jsx のCSS解析の改善
+
+Rust ベースのコンパイラの上に、styled-jsx Babel 変換に使用された CSS パーサをベースにした新しい CSS パーサを実装しました。
+この新しいパーサーは CSS の取り扱いを改善し、以前はすり抜けて予期せぬ動作を引き起こしていた無効な CSS が使用された場合にエラーを発生させるようになりました。
+
+この変更により、開発中および `next build` の際に無効な CSS がエラーを投げるようになります。この変更は、styled-jsx の使用にのみ影響します。
+
+### `next/image`のラップ要素が変更されました
+
+`next/image`は `<div>` の代わりに `<span>` で囲われた `<img>` をレンダーするようになりました。
+
+span を対象とした特定の CSS、たとえば `.container span` を使用している場合、Next.js 12 にアップグレードすると `<Image>` コンポーネント内のラップ要素に正しくマッチしない場合があります。
+これはセレクタを `.container span.item` のような特定のクラスに制限することや、関連事項するコンポーネントを `<span className="item" />` のような className で更新することで避けることができます。
+
```

Comment:
```suggestion
If your application has specific CSS targeting the `next/image` `<div>` tag, for example `.container div`, it may not match anymore. You can update the selector `.container span`, or preferably, add a new `<div className="wrapper">` wrapping the `<Image>` component and target that instead such as `.container .wrapper`.
```

ここの部分が抜けてるかもです(?)